tests/helpers/test_script.py

```diff
@@ -101,9 +104,12 @@ def assert_element(trace_element, expected_element, path):
     else:
         assert trace_element._error is None
 
-    # Don't check variables when script starts
+    # Ignore the context variable in the first step
     if trace_element.path == "0":
-        return
+        if variables := expected_element.get("variables"):
```

Comment:
Can we call `trace_element.as_dict` instead and just do a simple equality assertion of the trace element dict under test and the expected element? Then just parametrize the expected element.

The current code is really hard to follow.

Reply:
Yes, or even better, a custom syrupy serializer so we can just use snapshots.
Let's do that in a follow-up though.

Reply:
Ok.